python/plugins/processing/algs/qgis/Climb.py

```diff
@@ -0,0 +1,260 @@
+            if feedback.isCanceled():
+                break
+            climb = 0
+            descent = 0
+            minelev = float('Infinity')
+            maxelev = float('-Infinity')
+            # In case of multigeometries we need to do the parts
+            parts = feature.geometry().constParts()
+            partnumber = 0
+            if not feature.hasGeometry():
+                no_geometry.append(self.tr(
+                    'Feature: {feature_id}'.format(
+                        feature_id=feature.id())
+                )
+                )
+            for part in parts:
+                # Calculate the climb
+                first = True
+                zval = 0
+                for idx, v in enumerate(part.vertices()):
+                    zval = v.z()
+                    if math.isnan(zval):
+                        no_z_nodes.append(self.tr(
+                            'Feature: {feature_id}, part: {part_id}, point: {point_id}'.format(
+                                feature_id=feature.id(),
+                                part_id=partnumber,
+                                point_id=idx)
+                        )
+                        )
+                        continue
+                    if first:
+                        prevz = zval
+                        minelev = zval
+                        maxelev = zval
+                        first = False
+                    else:
+                        diff = zval - prevz
+                        if diff > 0:
+                            climb = climb + diff
+                        else:
+                            descent = descent - diff
+                        if minelev > zval:
+                            minelev = zval
+                        if maxelev < zval:
+                            maxelev = zval
+                    prevz = zval
+                totalclimb = totalclimb + climb
+                totaldescent = totaldescent + descent
+                partnumber += 1
+            # Set the attribute values
+            attrs = feature.attributes()
+            # Append the attributes to the end of the existing ones
+            attrs.append(climb)
+            attrs.append(descent)
+            attrs.append(minelev)
+            attrs.append(maxelev)
+
+            # Set the final attribute list
+            feature.setAttributes(attrs)
+            # Add a feature to the sink
+            sink.addFeature(feature, QgsFeatureSink.FastInsert)
+            if minelevation > minelev:
+                minelevation = minelev
+            if maxelevation < maxelev:
+                maxelevation = maxelev
+            # Update the progress bar
+            if fcount > 0:
+                feedback.setProgress(int(100 * current / fcount))
+
+        feedback.pushInfo(self.tr(
+            'The following features do not have geometry: {no_geometry_report}'.format(
+                no_geometry_report=str(no_geometry))
```

Comment:
```suggestion
                no_geometry_report=no_geometry)
```

Reply:
`no_geometry` is a list. AFAIK `feedback` only accepts strings, isn't it?

Reply:
So `', '.join(no_geometry)` ?

Reply:
still don't get why changing this..

Reply:
Because it's prettier ;-)
```python
>>>no_geometry=['geom1', 'geom2']
>>>print(no_geometry)
['geom1', 'geom2']
>>>print(', '.join(no_geometry))
geom1, geom2
```
Anyway, no problem.